deliverystream/aws-kinesisfirehose-deliverystream.json

```diff
@@ -755,6 +807,72 @@
           "type": "boolean"
         }
       }
+    },
+    "HttpEndpointRequestConfiguration": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "ContentEncoding": {
+          "type": "string",
+          "enum": [
+            "NONE",
+            "GZIP"
+          ]
+        },
+        "CommonAttributes": {
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "$ref": "#/definitions/HttpEndpointCommonAttribute"
+          },
+          "minItems": 0,
+          "maxItems": 50
+        }
+      }
+    },
+    "HttpEndpointCommonAttribute": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "AttributeName": {
```

Comment:
Would it be good to include the pattern here for `AttributeName` and `AttributeValue` fields? Also curious, do we include patterns here in the cfn resource type spec for all the fields where pattern is specified in the API reference or not? A quick search showed me atm we are only doing it for a subset of fields like `DeliveryStreamName` and `RoleArn` in various configurations.

Reply:
Most of them line up with the pattern specified in service model. For AttributeName we don't have restrict pattern in service model